packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -434,75 +416,308 @@ class AppBar extends StatelessWidget {
         data: appBarIconTheme,
         child: new DefaultTextStyle(
           style: sideStyle,
-          child: toolbar
-        )
-      )
+          child: toolbar,
+        ),
+      ),
     );
 
-    final double bottomOpacity = _bottomOpacity(size.height, statusBarHeight);
-    if (bottom != null) {
+
+    if (config.bottom != null) {
       appBar = new Column(
         mainAxisAlignment: MainAxisAlignment.spaceBetween,
         children: <Widget>[
           appBar,
-          bottomOpacity == 1.0 ? bottom : new Opacity(
-            child: bottom,
-            opacity: const Interval(0.25, 1.0, curve: Curves.fastOutSlowIn).transform(bottomOpacity)
-          )
-        ]
+          config.bottomOpacity == 1.0 ? config.bottom : new Opacity(
+            opacity: const Interval(0.25, 1.0, curve: Curves.fastOutSlowIn).transform(config.bottomOpacity),
+            child: config.bottom,
+          ),
+        ],
       );
     }
 
     // The padding applies to the toolbar and tabbar, not the flexible space.
-    // The incoming padding parameter's top value typically equals the height
-    // of the status bar - so that the toolbar appears below the status bar.
-    appBar = new Padding(
-      padding: padding,
-      child: appBar
-    );
-
-    // If the appBar's height shrinks below collapsedHeight, it will be clipped and bottom
-    // justified. This is so that the toolbar and the tabbar appear to move upwards as
-    // the appBar's height is reduced below collapsedHeight.
-    final double paddedCollapsedHeight = collapsedHeight + padding.top + padding.bottom;
-    if (size.height < paddedCollapsedHeight) {
-      appBar = new ClipRect(
-        child: new OverflowBox(
-          alignment: FractionalOffset.bottomLeft,
-          minHeight: paddedCollapsedHeight,
-          maxHeight: paddedCollapsedHeight,
-          child: appBar
-        )
+    if (config.primary) {
+      appBar = new Padding(
+        padding: MediaQuery.of(context).padding,
```

Comment:
You just want the top padding.  For example, if there's bottom padding, you don't want that.